### PR TITLE
HID-1804 - Added mailhog to docker-compose example file and nodemon

### PR DESCRIPTION
**Pull request by guillaumev**
We will have to see with @teodorescuserban if it's possible to run "npm start" on production and dev, but run "npm run dev" on our local environments to have it run with nodemon...

### CODE REVIEW

**Comment by rupl** [comment]
I do believe it's possible. For example on DSR we have a [`run_node.dev`](https://github.com/UN-OCHA/reports-site-stack/blob/master/env/local/run_node.dev) file in the stack which is intended for local development, but the infrastructure runs this [`run_node` file in the DSR repo](https://github.com/UN-OCHA/reports-site/blob/dev/run_node)

File: docker-compose.example.yml [hunk 1]
```diff
@@ -5,46 +5,24 @@ services:
...
-      SMTP_HOST: app.mailhog.vm
+      ROOT_URL: http://localhost:3000
+      SMTP_HOST: mailhog
       SMTP_PORT: 1025
-      SMTP_USER: humanitarianid
-      SMTP_PASS: smtp_password
-      APP_URL: http://app.hid.vm
+      SMTP_USER: 
+      SMTP_PASS:
+      APP_URL: https://app.hid.vm
```
Comment (teodorescuserban) [inline docker-compose.example.yml]: what is this for?

**Comment by teodorescuserban** [comment]
Indeed, you map your custom startup script as `/etc/services.d/node/run`.

File: docker-compose.example.yml (hunk 1 above)
Comment (guillaumev) [reply docker-compose.example.yml]: Some emails sent out by the API have a link which needs to point to the app URL, which is configured here

**Comment by guillaumev** [comment]
@rupl can you check this and make sure it doesn't break things on your side ?

File: docker-compose.example.yml [hunk 2]
```diff
@@ -5,46 +5,24 @@ services:
...
       NODE_ENV: local
-      ROOT_URL: http://api.hid.vm
-      SMTP_HOST: app.mailhog.vm
+      ROOT_URL: http://localhost:3000
+      SMTP_HOST: mailhog
       SMTP_PORT: 1025
-      SMTP_USER: humanitarianid
-      SMTP_PASS: smtp_password
-      APP_URL: http://app.hid.vm
+      SMTP_USER:
```
Comment (emmajane) [inline docker-compose.example.yml]: Does this intentionally remove a user/pass?

Comment (teodorescuserban) [reply docker-compose.example.yml]: it makes possible to actually use the SMTP_USER and SMTP_PASS in the `.env` file to define it for the desired environment. Providing a hardcoded default was useless anyway.

**Pull request merged**